pkg/provision/buildroot.go

```diff
@@ -70,14 +70,16 @@ func (p *BuildrootProvisioner) GenerateDockerOptions(dockerPort int) (*provision
 		klog.Infof("root file system type: %s", fstype)
 		noPivot = fstype == "rootfs"
 	}
-
+	/* Template is copied from https://github.com/moby/moby/blob/44bca1adf3c806c4ed6688d67c6f995653b09b26/contrib/init/systemd/docker.service#L2 with the following changes:
+	   After: minikube-automount.service is added
+	   ExecStart: additional flags are added
+	*/
 	engineConfigTmpl := `[Unit]
 Description=Docker Application Container Engine
 Documentation=https://docs.docker.com
-After=network.target  minikube-automount.service docker.socket
-Requires= minikube-automount.service docker.socket 
-StartLimitBurst=3
```

Comment:
@ComradeProgrammer lets keep 
StartLimitBurst=3
StartLimitIntervalSec=60

those are useful it wont let minikube's docker to restart 100 times in 1 second and cause CPU spikes...specially in CI enviroments with limiited resources